### PR TITLE
Add D97 Phase-2 geometry

### PR DESCRIPTION
**Pull request by srimanob**
#### PR description:
As requested in https://github.com/cms-sw/cmssw/pull/39571#discussion_r985528347, the geometry with T25 is brought back based on D88.

#### PR validation:
Tested with `24434.0 (2026D97+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal)`

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:
No need of backport

FYI @mmusich @emiglior

### CODE REVIEW

**Comment by srimanob** [comment]
test parameters:
- workflow = 24434.0

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by srimanob** [comment]
+Upgrade

The new D97 workflow runs fine. The error on PR test does not relate to this PR.

**Comment by srimanob** [comment]
@cmsbuild please test

Another try, to see if we can move away the relvals-input error.

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by srimanob** [comment]
+Upgrade

This PR adds D97 which is T25+D88 (baseline) as requests by Tracker. The new wfs run fine. We may add later T25+new baseline when we move.

**Comment by bsunanda** [comment]
+geometry

File: Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py [hunk 1]
```diff
@@ -2153,6 +2155,13 @@ def condition(self, fragment, stepList, key, hasHarvest):
         'Era' : 'Phase2C17I13M9',
         'ScenToRun' : ['GenSimHLBeamSpot','DigiTrigger','RecoGlobal', 'HARVESTGlobal'],
     },
+    '2026D97' : {
+        'Geom' : 'Extended2026D97',
+        'HLTmenu': '@fake2',
+        'GT' : 'auto:phase2_realistic_T25',
+        'Era' : 'Phase2C17I13M9',
+        'ScenToRun' : ['GenSimHLBeamSpot','DigiTrigger','RecoGlobal', 'HARVESTGlobal'],
```
Comment (mmusich) [inline Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:2163]: as this is waiting for signatures, do you mind adding the step `ALCAPhase2` here, as done for the other geometries in https://github.com/cms-sw/cmssw/pull/39858 ?

Comment (srimanob) [reply Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py]: Done.

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by srimanob** [comment]
+Upgrade

**Comment by srimanob** [comment]
@cms-sw/geometry-l2 @cms-sw/pdmv-l2 
Could you please consider, and sign/resign? Thanks.

**Comment by bsunanda** [comment]
+geometry

**Comment by kskovpen** [comment]
+pdmv

File: Configuration/Geometry/python/GeometryDD4hepExtended2026D97Reco_cff.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+from Geometry.ForwardGeometry.ForwardGeometry_cfi import *
+
+# timing
+from RecoMTD.DetLayers.mtdDetLayerGeometry_cfi import *
+from Geometry.MTDGeometryBuilder.mtdParameters_cff import *
+from Geometry.MTDNumberingBuilder.mtdNumberingGeometry_cff import *
+from Geometry.MTDNumberingBuilder.mtdTopology_cfi import *
+from Geometry.MTDGeometryBuilder.mtdGeometry_cfi import *
+from Geometry.MTDGeometryBuilder.idealForDigiMTDGeometry_cff import *
+mtdGeometry.applyAlignment = cms.bool(False)
```
Comment (perrotta) [inline Configuration/Geometry/python/GeometryDD4hepExtended2026D97Reco_cff.py:59]:
```suggestion
mtdGeometry.applyAlignment = False
```

File: Configuration/Geometry/python/GeometryExtended2026D97Reco_cff.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+from Configuration.Geometry.GeometryExtended2026D97_cff import *
+
+# tracker
+from Geometry.CommonTopologies.globalTrackingGeometry_cfi import *
+from RecoTracker.GeometryESProducer.TrackerRecoGeometryESProducer_cfi import *
+from Geometry.TrackerGeometryBuilder.TrackerAdditionalParametersPerDet_cfi import *
+from Geometry.TrackerGeometryBuilder.trackerParameters_cff import *
+from Geometry.TrackerNumberingBuilder.trackerTopology_cfi import *
+from Geometry.TrackerGeometryBuilder.idealForDigiTrackerGeometry_cff import *
+trackerGeometry.applyAlignment = cms.bool(False)
```
Comment (perrotta) [inline Configuration/Geometry/python/GeometryExtended2026D97Reco_cff.py:15]:
```suggestion
trackerGeometry.applyAlignment = False
```

File: Configuration/Geometry/python/GeometryExtended2026D97Reco_cff.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+from Geometry.ForwardGeometry.ForwardGeometry_cfi import *
+
+# timing
+from RecoMTD.DetLayers.mtdDetLayerGeometry_cfi import *
+from Geometry.MTDGeometryBuilder.mtdParameters_cff import *
+from Geometry.MTDNumberingBuilder.mtdNumberingGeometry_cff import *
+from Geometry.MTDNumberingBuilder.mtdTopology_cfi import *
+from Geometry.MTDGeometryBuilder.mtdGeometry_cfi import *
+from Geometry.MTDGeometryBuilder.idealForDigiMTDGeometry_cff import *
+mtdGeometry.applyAlignment = cms.bool(False)
```
Comment (perrotta) [inline Configuration/Geometry/python/GeometryExtended2026D97Reco_cff.py:59]:
```suggestion
mtdGeometry.applyAlignment = False
```

**Comment by srimanob** [comment]
Hi @perrotta 

Those lines are generated automatically from 
https://github.com/cms-sw/cmssw/blob/master/Configuration/Geometry/python/dict2026Geometry.py
i.e.
https://github.com/cms-sw/cmssw/blob/master/Configuration/Geometry/python/dict2026Geometry.py#L282

I should fix `dict2026Geometry.py` and regenerate all XML files. May I propose to merge this PR, and we do follow up PR, so it is clear on what PR for? Thanks.

**Comment by perrotta** [comment]
+1

**Comment by makortel** [comment]
For some reason the `test2026Geometry` unit test fails in the IB
```
===== Test "test2026Geometry" ====
cmsExtendedGeometry2026D97XML_cfi.py differs
cmsExtendedGeometry2026D97.xml differs

---> test test2026Geometry had ERRORS
```
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_amd64_gcc10/CMSSW_12_6_X_2022-11-04-1100/unitTestLogs/Configuration/Geometry#/

**Comment by srimanob** [comment]
Thanks @makortel 
That is because we merge https://github.com/cms-sw/cmssw/pull/39974 almost the same time, and missed D97 there. I will submit a fix, together with code cleaning as suggested in https://github.com/cms-sw/cmssw/pull/39841#discussion_r1010177965

**Comment by srimanob** [comment]
Fix in https://github.com/cms-sw/cmssw/pull/39992

**Pull request merged**